src/reflect/scala/reflect/internal/util/AbstractFileClassLoader.scala

```diff
@@ -76,15 +76,20 @@ class AbstractFileClassLoader(val root: AbstractFile, parent: ClassLoader)
     else
       defineClass(name, bytes, 0, bytes.length, protectionDomain)
   }
+
+  // on JDK 20 the URL constructor we're using is deprecated, but the recommended
+  // replacement, URL.of, doesn't exist on JDK 8
+  @annotation.nowarn("cat=deprecation")
   override protected def findResource(name: String): URL = findAbstractFile(name) match {
     case null => null
-    case file => URL.of(new URI(s"memory:${file.path}"), new URLStreamHandler {
+    case file => new URL(null, s"memory:${file.path}", new URLStreamHandler {
       override def openConnection(url: URL): URLConnection = new URLConnection(url) {
         override def connect() = ()
         override def getInputStream = file.input
       }
     })
   }
+
```

Comment:
Recently I proposed the compiler should infer EOF on a blank line following 2 or more vertical spacing braces, but I don't know if the rule applies when there is a paren.

Reply:
one of the things I like about the Scala 3 indentation-based syntax is that these sorts of questions don't even arise :shrug:

Reply:
indeed, Scala 3 syntax has only inflamed the provocations on Scala 2.